### PR TITLE
fix: runtime versions overwriting nuget versions

### PR DESCRIPTION
**Pull request by 37IulianPopovici**
This PR fixes some niche cases where the runtime versions are overwriting the correct nuget versions.

### CODE REVIEW

**Review by DOlufemi — APPROVED** [review]

File: lib/nuget-parser/parsers/dotnet-core-v2-parser.ts [hunk 1]
```diff
@@ -110,11 +110,31 @@ function getRestoredProjectName(
...
+      if (!value.serviceable && !value.sha512) {
+        localPackages.push(key);
+      }
+    }
+  }
+
+  return localPackages;
+}
+
+function versionStringToInt(version: string) {
```
Comment (dotkas) [inline lib/nuget-parser/parsers/dotnet-core-v2-parser.ts:127]: Maybe a comment with an example. Why is it needed? This looks like something that's going to fail when MS starts to do stuff differently. Is there a way around it?

File: lib/nuget-parser/parsers/dotnet-core-v2-parser.ts [hunk 2]
```diff
@@ -110,11 +110,31 @@ function getRestoredProjectName(
   );
 }
 
+export function extractLocalPackages(libs: Record<string, any>): string[] {
```
Comment (dotkas) [inline lib/nuget-parser/parsers/dotnet-core-v2-parser.ts:113]: Comment with explanation or example if possible. Not sure I get it. Why if not serviceable and not sha512?

**Review by dotkas — COMMENTED** [review]
Just comments.

File: test/parsers/parse-core-v2.spec.ts [hunk 3]
```diff
@@ -149,6 +157,7 @@ describe('when generating depGraphs and runtime assemblies using the v2 parser',
       const result = await plugin.inspect(projectPath, projectAssetsJson, {
         'dotnet-runtime-resolution': true,
         'dotnet-target-framework': targetFramework,
+        useFixForImprovedDotnetFalsePositives: true,
```
Comment (dotkas) [inline test/parsers/parse-core-v2.spec.ts:160]: Are you testing a negative case also?

Comment (37IulianPopovici) [reply test/parsers/parse-core-v2.spec.ts:160]: removed the FF

File: lib/nuget-parser/parsers/dotnet-core-v2-parser.ts (hunk 1 above)
Comment (37IulianPopovici) [reply lib/nuget-parser/parsers/dotnet-core-v2-parser.ts]: removed the method: not needed anymore

File: test/fixtures/dotnetcore/dotnet_6_local_package_reference/proj1/expected_depgraph.json [hunk 4]
```diff
@@ -489,10 +489,10 @@
         }
       },
       {
-        "id": "System.Text.Encodings.Web@6.0.0",
+        "id": "System.Text.Encodings.Web@4.7.2",
         "info": {
           "name": "System.Text.Encodings.Web",
-          "version": "6.0.0"
+          "version": "4.7.2"
```
Comment (37IulianPopovici) [inline test/fixtures/dotnetcore/dotnet_6_local_package_reference/proj1/expected_depgraph.json:495]: proj2.csproj [references](https://github.com/snyk/snyk-nuget-plugin/blob/main/test/fixtures/dotnetcore/dotnet_6_local_package_reference/proj2/proj2.csproj#L7) System.Text.Encodings.Web@4.7.2

**Review by dotkas — APPROVED** [review]

**Pull request merged**